wordpress-mobile/WordPress-Android
# Scan: Fetch scan state with polling

Parent: #13326

This PR 
- Adds use case to fetch scan state 
- Replaces fetching from `ScanStatusService` to newly added UseCase in the scan VM
- Copies progress state, VH from backup download implementation to common section for reuse in scan 
- Updates scan state builder to show progress item for SCANNING scan state

Start Scan Child PR: #13741

**To test**

Prerequisite: 

- Make sure both Scan feature flag and MySiteImprovements flag are set to on in the App settings.
- WP.com account with a site having scan capability (e.g. https://pressable-jetpack-daily-scan.mystagingwebsite.com).

1. Login to the app using above WP.com account and select the site on the My Site tab
2. Click scan menu item on My Site
3. Notice that scan state is fetched and shown
4. Return to My Site screen
5. Start scan for the site from the browser (Login to WP.com account on the web browser, select the site  -> Jetpack -> Scan, click "Scan now" button)
6. Click scan menu item on My Site again while scanning is in progress
7. Notice that scanning scan state is fetched and shown with a progress bar (progress updates shown in a label)
(Tests included in `FetchScanStateUseCaseTest` and `ScanViewModelTest` )

**Notes**

1. Ignore all styling
2. (Initial) Loading/ Failure states not handled
3. There's a small delay before UI is displayed on screen load (while first fetch is not complete). Showing last scan state for the site from the db (before first fetch) can be done either using a separate `GetScanStateUseCase` or from inside the `FetchScanStateUseCase`.

**Merge Instructions**

1. Review FluxC PR https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1833, merge and create a tag in FluxC develop
2. Replace FluxC hash with tag from develop 
2. Remove the "Not Ready for Merge label"
3. Merge this PR

PR submission checklist:
